# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/llvm/hotfix-sparcv9-not-64.patch
# Copyright (C) 2024 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- llvm-19.1.3.src/lib/TargetParser/Triple.cpp.orig	2024-10-29 11:09:39.000000000 +0100
+++ llvm-19.1.3.src/lib/TargetParser/Triple.cpp	2024-11-18 15:16:46.152757300 +0100
@@ -68,7 +68,7 @@
   case shave:          return "shave";
   case sparc:          return "sparc";
   case sparcel:        return "sparcel";
-  case sparcv9:        return "sparcv9";
+  case sparcv9:        return "sparc64";
   case spir64:         return "spir64";
   case spir:           return "spir";
   case spirv:          return "spirv";
@@ -423,8 +423,10 @@
     .Case("riscv64", riscv64)
     .Case("hexagon", hexagon)
     .Case("sparc", sparc)
+    .Case("sparcv8", sparc)
+    .Case("sparcv9", sparc)
     .Case("sparcel", sparcel)
-    .Case("sparcv9", sparcv9)
+    .Case("sparc64", sparcv9)
     .Case("s390x", systemz)
     .Case("systemz", systemz)
     .Case("tce", tce)
@@ -571,9 +573,9 @@
           .Case("riscv64", Triple::riscv64)
           .Case("hexagon", Triple::hexagon)
           .Cases("s390x", "systemz", Triple::systemz)
-          .Case("sparc", Triple::sparc)
+          .Cases("sparc", "sparcv8", "sparcv9", Triple::sparc)
           .Case("sparcel", Triple::sparcel)
-          .Cases("sparcv9", "sparc64", Triple::sparcv9)
+          .Case("sparc64", Triple::sparcv9)
           .Case("tce", Triple::tce)
           .Case("tcele", Triple::tcele)
           .Case("xcore", Triple::xcore)
